packages/batch-submitter/src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -742,70 +706,41 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
     })
-    const txType = block.transactions[0].txType
+
+    const sequencerFields = {
+      isSequencerTx: false,
+      sequencerTxType: undefined,
+      rawTransaction: undefined,
+    }
 
     if (this._isSequencerTx(block)) {
-      if (txType === TxType.EIP155 || txType === TxType.EthSign) {
-        return this._getDefaultEcdsaTxBatchElement(block)
-      } else {
-        throw new Error('Unsupported Tx Type!')
-      }
-    } else {
-      return {
-        stateRoot: block.stateRoot,
-        isSequencerTx: false,
-        sequencerTxType: undefined,
-        txData: undefined,
-        timestamp: block.timestamp,
-        blockNumber: block.transactions[0].l1BlockNumber,
-      }
+      sequencerFields.isSequencerTx = true
+      sequencerFields.sequencerTxType =
+        txTypePlainText[block.transactions[0].txType]
+      sequencerFields.rawTransaction = block.transactions[0].rawTransaction
     }
-  }
 
-  private async _getBlock(blockNumber: number): Promise<L2Block> {
-    const block = (await this.l2Provider.getBlockWithTransactions(
-      blockNumber
-    )) as L2Block
-    // Convert the tx type to a number
-    block.transactions[0].txType = txTypePlainText[block.transactions[0].txType]
-    block.transactions[0].queueOrigin =
-      queueOriginPlainText[block.transactions[0].queueOrigin]
-    if (!block.transactions[0].l1BlockNumber) {
-      this.log.error('Current block missing l1BlockNumber', {
-        blockNumber,
-        block,
-      })
+    const batchElement = {
+      ...sequencerFields,
+      stateRoot: block.stateRoot,
+      timestamp: formatNumber(block.timestamp),
+      blockNumber: formatNumber(block.transactions[0].l1BlockNumber),
     }
 
-    return block
+    return batchElement
   }
 
-  private _getDefaultEcdsaTxBatchElement(block: L2Block): BatchElement {
-    const tx: TransactionResponse = block.transactions[0]
-    const txData: EIP155TxData = {
-      sig: {
-        v: tx.v - this.l2ChainId * 2 - 8 - 27,
-        r: tx.r,
-        s: tx.s,
-      },
-      gasLimit: BigNumber.from(tx.gasLimit).toNumber(),
-      gasPrice: BigNumber.from(tx.gasPrice).toNumber(),
-      nonce: tx.nonce,
-      target: tx.to ? tx.to : '00'.repeat(20),
-      data: tx.data,
-      type: block.transactions[0].txType,
-    }
-    return {
-      stateRoot: block.stateRoot,
-      isSequencerTx: true,
-      sequencerTxType: block.transactions[0].txType,
-      txData,
-      timestamp: block.timestamp,
-      blockNumber: block.transactions[0].l1BlockNumber,
-    }
+  private async _getBlock(blockNumber: number): Promise<L2Block> {
+    return (await this.l2Provider.send('eth_getBlockByNumber', [
+      toRpcHexString(blockNumber),
+      true,
+    ])) as L2Block
   }
 
   private _isSequencerTx(block: L2Block): boolean {
-    return block.transactions[0].queueOrigin === QueueOrigin.Sequencer
+    return (
+      queueOriginPlainText[block.transactions[0].queueOrigin] ===
+      QueueOrigin.Sequencer
+    )
```

Comment:
Why is this needed? This should be getting set [here](https://github.com/ethereum-optimism/optimism/blob/master/packages/core-utils/src/l2context.ts#L34-L35) already, no need for plaintext stuff?

Reply:
^i think this may no longer be needed with `injectL2Context`, which wasn't used the last time we made this change to batch submitter

Reply:
Removed all the crappy plaintext things